x/distribution/keeper/allocation.go

```diff
@@ -52,34 +53,52 @@ func (k Keeper) AllocateTokens(ctx context.Context, totalPreviousPower int64, bo
 	feeMultiplier := feesCollected.MulDecTruncate(voteMultiplier)
 
 	// allocate tokens proportionally to voting power
-	//
-	// TODO: Consider parallelizing later
-	//
-	// Ref: https://github.com/cosmos/cosmos-sdk/pull/3099#discussion_r246276376
+	var g errgroup.Group
+	// channel for remaining rewards
+	remainingCh := make(chan sdk.DecCoins, len(bondedVotes))
+
 	for _, vote := range bondedVotes {
+		vote := vote
 
-		validator, err := k.stakingKeeper.ValidatorByConsAddr(ctx, vote.Validator.Address)
-		if err != nil {
-			return err
-		}
+		g.Go(func() error {
```

Comment:
could we have some benchmarks on this?

Reply:
could we get a short fuzz test for this function, if possible?

Reply:
I agree it would be good to benchmark single threaded vs concurrent.   the results might be surprising.  iavl v0 & v1 is a single threaded reader, so if this func is i/o bound, there may not be too much wall clock improvement, in fact the coordination overhead could make things slower.   iavl v2 scales well with concurrent reads, not sure that's relevant here though.

Reply:
sure, will add 👍🏾